pkg/controller/replicaset/replica_set.go

```diff
@@ -198,6 +219,19 @@ func (rsc *ReplicaSetController) Run(workers int, stopCh <-chan struct{}) {
 // getPodReplicaSet returns the replica set managing the given pod.
 // TODO: Surface that we are ignoring multiple replica sets for a single pod.
 func (rsc *ReplicaSetController) getPodReplicaSet(pod *api.Pod) *extensions.ReplicaSet {
+	// look up in the cache, if cached and the cache is valid, just return cached value
+	obj, cached := rsc.lookupCache.GetMatchingObject(pod)
+	rs, ok := obj.(*extensions.ReplicaSet)
+	if !ok {
```

Comment:
> So basically:
rsc.lookupCache.GetMatchingObject(pod) is returning nil

if `rsc.lookupCache.GetMatchingObject(pod) is returning nil` returning nil, we must add a nil pointer into the cache somewhere, right? Then, when did we add a nil pointer into the cache?